js/node/src/session_options_helper.cc

```diff
@@ -111,9 +126,14 @@ void ParseExecutionProviders(const Napi::Array epList, Ort::SessionOptions& sess
 #endif
 #ifdef USE_QNN
     } else if (name == "qnn") {
-      std::unordered_map<std::string, std::string> qnn_options;
-      qnn_options["backend_path"] = "QnnHtp.dll";
-      qnn_options["enable_htp_fp16_precision"] = "1";
+      // Ensure that the backend_path and enable_htp_fp16_precision options are set to default values if not provided.
+      if (qnn_options.find("backend_path") == qnn_options.end()) {
+        qnn_options["backend_path"] = "QnnHtp.dll";
+      }
+      if (qnn_options.find("enable_htp_fp16_precision") == qnn_options.end()) {
```

Comment:
QNN EP already set enable_htp_fp16_precision to 1 by default, so you can skip it here.

Reply:
Good call. Removed the code setting enable_htp_fp16_precision flag here as such.